components/chassis.py

```diff
@@ -61,67 +58,75 @@ def __init__(
         self.drive_id = drive_id
-        self.encoder = phoenix5.sensors.CANCoder(encoder_id)
+        self.encoder = phoenix6.hardware.CANcoder(encoder_id)
 
         # Reduce CAN status frame rates before configuring
-        self.steer.setStatusFramePeriod(
-            phoenix5.StatusFrameEnhanced.Status_1_General, 250, 10
-        )
-        self.drive.setStatusFramePeriod(
-            phoenix5.StatusFrameEnhanced.Status_1_General, 250, 10
-        )
+        self.steer.get_fault_field().set_update_frequency(4, 0.01)
+        self.drive.get_fault_field().set_update_frequency(4, 0.01)
 
         # Configure steer motor
-        self.steer.setNeutralMode(phoenix5.NeutralMode.Brake)
-        self.steer.setInverted(steer_reversed)
-        self.steer.config_kP(0, 0.15035, 10)
-        self.steer.config_kI(0, 0, 10)
-        self.steer.config_kD(0, 5.6805, 10)
-        self.steer.configAllowableClosedloopError(
-            0, self.STEER_RAD_TO_COUNTS * math.radians(4)
-        )
-        self.steer.configSelectedFeedbackSensor(
-            phoenix5.FeedbackDevice.IntegratedSensor, 0, 10
-        )
+        steer_config = self.steer.configurator
+
+        steer_motor_config = phoenix6.configs.MotorOutputConfigs()
+        steer_motor_config.neutral_mode = phoenix6.signals.NeutralModeValue.BRAKE
+        steer_motor_config.inverted = steer_reversed
+
+        steer_pid = phoenix6.configs.Slot0Configs()  # configuration for motor pid
+        steer_pid.k_p = 0.15035
+        steer_pid.k_i = 0
+        steer_pid.k_d = 5.6805
+
+        steer_config.apply(steer_motor_config)
+        steer_config.apply(steer_pid, 0.01)
 
         # Configure drive motor
-        self.drive.setNeutralMode(phoenix5.NeutralMode.Brake)
-        self.drive.setInverted(drive_reversed)
-        self.drive.configVoltageCompSaturation(self.MAX_DRIVE_VOLTS, timeoutMs=10)
-        self.drive.enableVoltageCompensation(True)
-        self.drive_ff = SimpleMotorFeedforwardMeters(kS=0.18877, kV=2.7713, kA=0.18824)
-        self.drive.configVelocityMeasurementPeriod(
-            phoenix5.sensors.SensorVelocityMeasPeriod.Period_5Ms
-        )
-        self.drive.configVelocityMeasurementWindow(8)
-        self.drive.config_kP(0, 0.011489, 10)
-        self.drive.config_kI(0, 0, 10)
-        self.drive.config_kD(0, 0, 10)
+        drive_config = self.drive.configurator
+
+        drive_motor_config = phoenix6.configs.MotorOutputConfigs()
+        drive_motor_config.neutral_mode = phoenix6.signals.NeutralModeValue.BRAKE
+        drive_motor_config.inverted = steer_reversed
+
+        self.drive_pid_ff = (
+            phoenix6.configs.Slot1Configs()
+        )  # configuration for motor pid and feedforward
+        self.drive_pid_ff.k_p = 0.011489
+        self.drive_pid_ff.k_i = 0
+        self.drive_pid_ff.k_d = 0
+        self.drive_pid_ff.k_s = 0.18877
+        self.drive_pid_ff.k_v = 2.7713
+        self.drive_pid_ff.k_a = 0.18824
```

Comment:
The feedforward constants used metres as the distance base unit, but the PID gains used Talon units. Are these in the correct units for what Phoenix 6 expects?

Reply:
Don't we have to retune the pid and the ff anyways

Reply:
Yes, but we shouldn't destroy the modules because we had nonsense numbers either.

Reply:
alr I have converted the PID value to the new unit and I have reverted back to using wpimath ff as with the talon config ff idk how to even apply the ff values